test/cri-containerd/main.go

```diff
@@ -246,3 +250,30 @@ func findTestNvidiaGPUDevice() (string, error) {
 	}
 	return strings.TrimSpace(string(out)), nil
 }
+
+// Generates a gmsa credential spec and places it at 'path'. Dir has to exist, it will not be created by
+// New-CredentialSpec.
+func generateCredSpec(path, account string) error {
+	cmd := fmt.Sprintf(`New-CredentialSpec -AccountName %s -Path "%s"`, account, path)
+	_, err := exec.Command(
+		"powershell",
+		cmd,
+	).Output()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Tries to generate a cred spec to use for gmsa test cases. Returns the cred
+// spec and an error if any.
+func gmsaSetup(path, account string) (string, error) {
```

Comment:
Seems like this function should handle creating its own temp file instead of requiring the test to pass it. It looks like that value isn't used again in the tests.